docs/tutorials/pv_optimization.md

```diff
@@ -0,0 +1,125 @@
+# Optimizing PV production
+
+## Introduction
+
+Before we start it's assumed that you have finished the first [tutorial](./getting_started.md)
+
+In this tutorial you will write an application that optimizes the energy produced from a
+[PV](intro/glossary/#pv) system with Battery for self consumption.
+In order to do so you need to measure the power that flows through the
+[grid connection point](../../intro/glossary/#grid) to determine excess power.
+
+## Measure the excess power
+
+When using the term excess power what we actually mean is the consumer excess power, that is the power that
+flows from the PV system into the grid.
+
+!!! note
+
+    We are using the [passive sign convention](../../intro/glossary/#passive-sign-convention) and thus power
+    flowing from PV is negative and consumed power is positive.
+
+We want to measure the excess power. In order to do so you can use the SDK's data pipeline and especially
+the pre defined
+[consumer](../../reference/frequenz/sdk/timeseries/logical_meter/#frequenz.sdk.timeseries.logical_meter.LogicalMeter.consumer_power)
+and
+[producer power](../../reference/frequenz/sdk/timeseries/logical_meter/#frequenz.sdk.timeseries.logical_meter.LogicalMeter.producer_power)
+formulas.
+
+```python
+async def run() -> None:
+    ... # (1)!
+
+    # negative means feed-in power due to the negative sign convention
+    consumer_excess_power_engine = (
+        microgrid.logical_meter().producer_power
+        + microgrid.logical_meter().consumer_power
+    ).build("excess_power") # (2)!
+    cons_excess_power_recv = consumer_excess_power_engine.new_receiver() # (3)!
+```
+
+1. The initialization code as explained in the Getting Started tutorial.
+2. Construct the consumer excess power by summing up consumer and producer power each of which having
+opposite signs due to the sign convention. This returns a formula engine.
+3. Request a receiver from the formula engine which will be used to consume the stream.
+
+## Control the Battery
+
+Now, with the constructed excess power data stream use a
+[battery pool](../../reference/frequenz/sdk/timeseries/battery_pool/) to implement control logic.
+
+```python
+    ...
+
+    battery_pool = microgrid.battery_pool() # (1)!
+
+    async for cons_excess_power in cons_excess_power_recv: # (2)!
+        cons_excess_power = cons_excess_power.value # (3)!
+        if cons_excess_power is None: # (4)!
+            continue
+        if cons_excess_power <= Power.zero(): # (5)!
+            await battery_pool.charge(-cons_excess_power)
+        elif cons_excess_power > Power.zero():
```

Comment:
oh yes, the `charge`/`discharge` methods are no longer available, need to be replaced by the new `propose_*` ones.

Reply:
I believe it isn't hard, but it will probably need some reading up on how exactly do do it, maybe half a days work?

Reply:
I think it might be worth it now that we have more examples in `docs/`.

I would also like to resurrect the idea about just extracting examples to separate files and run all the tools as normal on them. Another advantage of this is we need to run `pytest` in all combinations of python and archs, but linting could be done only once, there is no need to run the linting on all python versions and archs, and in particular testing examples is super slow, so it makes test in arm64 take much longer than needed.

But the later is probably a fair amount of extra work, so we can also do it as a second step and first only focus on getting `docs/` tested.

Reply:
Let's continue in https://github.com/frequenz-floss/frequenz-repo-config-python/issues/136